evm_loader/program/src/query.rs

```diff
@@ -0,0 +1,134 @@
+//! `EVMLoader` query account cache.
+
+use std::collections::BTreeMap;
+use serde::{Deserialize, Serialize};
+use solana_program::{clock::Epoch, pubkey::Pubkey};
+use crate::solana_backend::AccountStorageInfo;
+
+const KB: usize = 1024;
+pub const MAX_CHUNK_LEN: usize = 8 * KB;
+
+/// Represents error states of queries.
+#[derive(thiserror::Error, Debug)]
+pub enum Error {
+    #[error("account not found")]
+    AccountNotFound,
+    #[error("invalid argument")]
+    InvalidArgument,
+}
+
+/// Result type for queries.
+pub type Result<T> = std::result::Result<T, Error>;
+
+/// Represents cache of queries to Solana accounts.
+#[derive(Serialize, Deserialize, Default, Debug)]
+pub struct AccountCache {
+    cache: BTreeMap<Pubkey, Value>,
+}
+
+impl Drop for AccountCache {
+    fn drop(&mut self) {
+        debug_print!("drop AccountCache");
+    }
+}
+
+impl AccountCache {
+    /// Creates new instance of the cache object.
+    pub fn new() -> Self {
+        debug_print!("new AccountCache");
+        Self {
+            cache: BTreeMap::new(),
+        }
+    }
+
+    /// Inserts or replaces entry into the cache.
+    pub fn put(&mut self, address: Pubkey, value: Value) {
+        self.cache.remove(&address);
+        self.cache.insert(address, value);
```

Comment:
`BTreeMap::insert` will overwrite an existing value.
`.remove()` is unnecessary.

Reply:
Indeed. Deleted the remove call.